app/models/grda_warehouse/synthetic/assessment.rb

```diff
@@ -1,12 +1,85 @@
+###
+# Copyright 2016 - 2021 Green River Data Analysis, LLC
+#
+# License detail: https://github.com/greenriver/hmis-warehouse/blob/production/LICENSE.md
+###
+
 module GrdaWarehouse::Synthetic
   class Assessment < GrdaWarehouseBase
     self.table_name = 'synthetic_assessments'
 
     belongs_to :enrollment, class_name: 'GrdaWarehouse::Hud::Enrollment'
     belongs_to :client, class_name: 'GrdaWarehouse::Hud::Client'
     belongs_to :source, polymorphic: true
+    belongs_to :hud_assessment, class_name: 'GrdaWarehouse::Hud::Assessment', optional: true
 
     validates_presence_of :enrollment
     validates_presence_of :client
+
+    # Subclasses must define:
+    #   assessment_date, assessment_type, assessment_level, prioritization_status, data_source
+
+    # Subclasses may override
+    def assessment_location
+      ''
+    end
+    alias AssessmentLocation assessment_location
+
+    def self.hud_sync
+      # Import synthetic events
+      GrdaWarehouse::Synthetic.available_assessment_types.each do |class_name|
+        class_name.constantize.sync
+      end
+
+      create_hud_assessments
+
+      # Clean up orphans in HUD table
+      GrdaWarehouse::Hud::Assessment.
+        where(synthetic: true).
+        where.not(id: select(:hud_assessment_id)).
+        delete_all
+    end
+
+    def self.create_hud_assessments
+      find_in_batches do |batch|
```

Comment:
I _think_ we want `preload(:enrollment, :client)` in here.

Reply:
Apparently, `preload` is supposed for polymorphic associations, but not `eager_load`...
https://api.rubyonrails.org/v5.1/classes/ActiveRecord/EagerLoadPolymorphicError.html